server/src/main/java/org/elasticsearch/snapshots/SnapshotShardsService.java

```diff
@@ -868,4 +876,76 @@ public void onFailure(Exception e) {
             )
         );
     }
+
+    private final ConsistencyChecker consistencyChecker = new ConsistencyChecker();
+
+    /**
+     * After receiving an ack from the master confirming that it marked a shard snapshot as complete, checks the local cluster state to
+     * confirm that it was updated to reflect this and logs a message if not. Not 100% watertight because this node might be lagging behind
+     * and hasn't received the cluster state yet, or else it might have been removed from the cluster, but both of these things will also
+     * be visible in the logs.
```

Comment:
How do we see (in logs) that the cluster state update hasn't been received yet?

Reply:
Added references in 765edb4ec70f, does that clarify?

Reply:
Yes, very helpful